go/net/formatted-template-string.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+    "fmt"
+    "html/template"
+    "net/http"
+)
+
+func Fine(r *http.Request) template.HTML {
+    // ok
+    return template.HTML("<html><body><h1>Hello, world</h1></body></html>")
+}
+
+func Concat(r *http.Request) template.HTML {
+    customerId := r.URL.Query().Get("id")
+    // ruleid: formatted-template-string
+    tmpl := "<html><body><h1>" + customerId + "</h1></body></html>"
+
+    return template.HTML(tmpl)
+}
+
+func ConcatInline(r *http.Request) template.HTML {
+    customerId := r.URL.Query().Get("id")
+
+    // ruleid: formatted-template-string
+    return template.HTML("<html><body><h1>" + customerId + "</h1></body></html>")
+}
+
+func ConcatInline(r *http.Request) template.HTML {
+    customerId := r.URL.Query().Get("id")
+
+    // ruleid: formatted-template-string
+    return template.HTML("<html><body><h1>" + customerId)
+}
+
+func Formatted(r *http.Request) template.HTML {
+    customerId := r.URL.Query().Get("id")
+    // todoruleid: formatted-template-string
+    tmpl, err := fmt.Printf("<html><body><h1>%s</h1></body></html>", customerId)
+    if (err != nil) {
+        return template.HTML("")
+    }
+    return template.HTML(tmpl)
+}
+
+func FormattedInline(r *http.Request) template.HTML {
+    customerId := r.URL.Query().Get("id")
+    // ruleid: formatted-template-string
+    return template.HTML(fmt.Sprintf("<html><body><h1>%s</h1></body></html>", customerId))
```

Comment:
Is there no good way to detect that `customerId` was assigned out of the `Request` object?

Reply:
In other checks, I used `pattern-inside` to limit checking when `$R *http.Request` was a function argument. I wanted to see what else was out there first: https://dev.massive.ret2.co/triager/filter/716